mantidproject/mantid
# Ellipsoidal integration in IntegratePeaksMD

Option to integrate an ellipsoid (instead of sphere) in IntegratePeaksMD with axes proportional to the sqrt of the eigenvalues of the covariance matrix (with the radius along the eigenvector with the largest eigenvalue equal to peak radius. 

There is an additional option to fix the ellipsoid axes to be parallel to Q of the peak.

**To test:**
(1) ask me for the MD and peak workspace nexus files
(2) run the script in mantidplot
(3) inspect the results in the sliceviewer for the spherical vs ellipsoidal integration. You should expect the ellipsoid to represent the peak shape fairly well (unless there is an anomaly like powder line in the background region etc.) see the image below
![image](https://user-images.githubusercontent.com/55979119/81150225-75e45580-8f77-11ea-97a5-98718c9f6b20.png)
(4) try switching the fixQAxis flag on/off - it shouldn't have too much of an effect on most peaks as the principal axis of the ellipsoid tends to be close to Q anyway.

```
from mantid.simpleapi import *

WISH00045319MD = LoadMD(Filename='WISH00045319MD.nxs')
peaks = LoadNexus(Filename='WISH00045319_peaks.nxs')
IntegratePeaksMD(InputWorkspace='WISH00045319MD', PeakRadius=0.15, BackgroundInnerRadius=0.15,
    BackgroundOuterRadius=0.2, Ellipsoid=True, PeaksWorkspace='peaks', OutputWorkspace='peaks_int', 
    UseOnePercentBackgroundCorrection=False)
IntegratePeaksMD(InputWorkspace='WISH00045319MD', PeakRadius=0.15, BackgroundInnerRadius=0.15,
    BackgroundOuterRadius=0.2, Ellipsoid=False, PeaksWorkspace='peaks', OutputWorkspace='peaks_int_sphere', 
    UseOnePercentBackgroundCorrection=False)
```

Fixes #27776 and #28339 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
